integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py

```diff
@@ -157,3 +156,65 @@ def delete_documents(self, document_ids: List[str]) -> None:
         if not document_ids:
             return
         self.collection.delete_many(filter={"id": {"$in": document_ids}})
+
+    def embedding_retrieval(
+        self,
+        query_embedding: List[float],
+        filters: Optional[Dict[str, Any]] = None,
+        top_k: int = 10,
+    ) -> List[Document]:
+        """
+        Find the documents that are most similar to the provided `query_embedding` by using a vector similarity metric.
+
+        :param query_embedding: Embedding of the query
+        :param filters: Optional filters.
+        :param top_k: How many documents to return.
+        """
+        if not query_embedding:
+            msg = "Query embedding must not be empty"
+            raise ValueError(msg)
+
+        query_embedding_np = np.array(query_embedding).astype(np.float32)
```

Comment:
Do we really need `numpy` in this case? We're converting back to a list right after, can we skip this or do it in plain Python?

Reply:
I was thinking of this as a "normalization" step - whatever type you're sending in, be it int or another type of float, this line brings it back to float32. Whether it's worth the cost idk, but I believe it can make this component more resilient. Wdyt?

Reply:
I don't see the reason to be fair. We state in the type hint that it must be a list of floats. 
Passing a list of ints doesn't make much sense either but if the users wants to do so let them. They can deal with the consequences of that decision.
Also if this will be called most of the times by the embedding retriever in a Pipeline, it will be pretty hard creating a connection that with a list of ints and a list of floats.